### PR TITLE
python3Packages.recoll: init

### LINKED ISSUE #118514: Recoll - Python API
**Project description**
The recoll package exposes a python API. It'd be awesome to have a package for that.

**Metadata**

* homepage URL: https://www.lesbonscomptes.com/recoll/ 
* related: https://www.lesbonscomptes.com/recoll/usermanual/webhelp/docs/RCL.INSTALL.BUILDING.PYTHON.html
https://packages.debian.org/stretch/python-recoll
* source URL: Main package already exists - https://github.com/NixOS/nixpkgs/blob/nixos-20.09/pkgs/applications/search/recoll/default.nix#L67 
* platforms: unix

### PR DESCRIPTION
**Pull request by dotlambda**
Also move some stuff to recoll's nativeBuildInputs.

###### Motivation for this change
fixes https://github.com/NixOS/nixpkgs/issues/118514

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@fabianhjr @karthikiyengar Do you want to be added to recoll's maintainers?

### CODE REVIEW

**Comment by karthikiyengar** [comment]
Thanks for such a quick PR. Python isn't my primary language and the implementation details are a bit unclear to me, but sure, I can sign up and figure things out along the way :-)

I tested it out with a `shell.nix` and it seems to work fine.
```
let
  pkgs = import <nixpkgs> {};
in pkgs.mkShell {
  buildInputs = [
    pkgs.python3
    (pkgs.libsForQt5.callPackage ./pkgs/applications/search/recoll {})
  ];
}
```

**Review by karthikiyengar — APPROVED** [review]

**Comment by dotlambda** [comment]
@karthikiyengar I would recommend you put
```
with import ./. { };

(python3.withPackages (ps: with ps; [ recoll ])).env
```
in your `shell.nix`. It's important that recoll uses the same version of Python as the one the executable called python refers to.

File: pkgs/applications/search/recoll/default.nix [hunk 1]
```diff
@@ -48,10 +48,12 @@ mkDerivation rec {
...
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [
+    file pkg-config python3Packages.setuptools which
+  ];
 
-  buildInputs = with python3Packages; [
-    bison chmlib file python setuptools which xapian zlib
+  buildInputs = [
+    bison chmlib python3Packages.python xapian zlib
```
Comment (SuperSandro2000) [inline pkgs/applications/search/recoll/default.nix:56]:
```suggestion
    bison chmlib python3 xapian zlib
```

File: pkgs/applications/search/recoll/default.nix [hunk 2]
```diff
@@ -48,10 +48,12 @@ mkDerivation rec {
     ++ lib.optionals (!withGui) [ "--disable-qtgui" "--disable-x11mon" ]
     ++ (if stdenv.isLinux then [ "--with-inotify" ] else [ "--without-inotify" ]);
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [
+    file pkg-config python3Packages.setuptools which
```
Comment (SuperSandro2000) [inline pkgs/applications/search/recoll/default.nix:52]:
```suggestion
    file pkg-config python3.pkgs.setuptools which
```

**Review by SuperSandro2000 — COMMENTED** [review]
I think we want to use python3 as an input here.

File: pkgs/applications/search/recoll/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/applications/search/recoll/default.nix:56]: I'd rather keep `python3Packages` so people overriding it don't have to change anything.

Comment (SuperSandro2000) [reply pkgs/applications/search/recoll/default.nix:56]: Is that a common thing to do with this package? Maybe we should upstream some of those overrides if it is a very common use case.

Comment (dotlambda) [reply pkgs/applications/search/recoll/default.nix:56]: No idea, I've never used it.

**Pull request merged**